internal/controller/role/role.go

```diff
@@ -0,0 +1,353 @@
+			role.SetConditions(xpv1.Available())
+		}
+	}
+
+	return managed.ExternalObservation{
+		// Return false when the external resource does not exist. This lets
+		// the managed resource reconciler know that it needs to call Create to
+		// (re)create the resource, or that it has successfully been deleted.
+		ResourceExists: exists,
+
+		// Return false when the external resource exists, but it not up to date
+		// with the desired managed resource state. This lets the managed
+		// resource reconciler know that it needs to call Update.
+		ResourceUpToDate: upToDate,
+
+		// Return any details that may be required to connect to the external
+		// resource. These will be stored as the connection secret.
+		ConnectionDetails: managed.ConnectionDetails{},
+	}, nil
+}
+
+// Create an AWS Secret Backend Role
+func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.ExternalCreation, error) {
+	role, ok := mg.(*v1alpha1.Role)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errNotRole)
+	}
+
+	err := c.writeRole(role)
+	if err != nil {
+		return managed.ExternalCreation{
+			ExternalNameAssigned: false,
+			ConnectionDetails:    managed.ConnectionDetails{},
+		}, errors.Wrap(err, errCreation)
+	}
+
+	return managed.ExternalCreation{
+		// Optionally return any details that may be required to connect to the
+		// external resource. These will be stored as the connection secret.
+		ConnectionDetails: managed.ConnectionDetails{},
+	}, nil
+}
+
+// Update an AWS Secret Backend Role - Thats the same approach we use for create it.
+func (c *external) Update(ctx context.Context, mg resource.Managed) (managed.ExternalUpdate, error) {
+	role, ok := mg.(*v1alpha1.Role)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errNotRole)
+	}
+
+	err := c.writeRole(role)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
+
+	return managed.ExternalUpdate{
+		// Optionally return any details that may be required to connect to the
+		// external resource. These will be stored as the connection secret.
+		ConnectionDetails: managed.ConnectionDetails{},
+	}, nil
+}
+
+// Delete an AWS Secret Backend Role
+func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
+	role, ok := mg.(*v1alpha1.Role)
+	if !ok {
+		return errors.New(errNotRole)
+	}
+
+	name := role.Name
+	authBackend := role.Spec.ForProvider.Backend
+
+	c.logger.Debug("Deleting role %q on AWS backend %q", name, authBackend)
+	_, err := c.client.Logical().Delete(authBackend + "/roles/" + name)
+	if err != nil {
+		return errors.Wrap(err, errDelete)
+	}
+	c.logger.Debug("Deleted role %q on AWS backend %q", name, authBackend)
+
+	return nil
+}
+
+// validate the role as some fields are only allowed in case another field has a different value
+//
+//nolint:gocyclo // Disabling lint here as we are doing many validation between fields that could not be done using kubebuilder
+func validate(role *v1alpha1.Role) error {
```

Comment:
I would put this as a method in the `CrossplaneToVault` with several "private" methods checking each field. THis you'd reduce the complexity lint issue.